### PR TITLE
Add support for specifying sector size for LUKS 2 devices

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Resolves: rhbz#1855726

### CODE REVIEW

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Comment by vojtechtrefny** [comment]
> My only question is whether we want/need to expose this in the factory. Is someone hoping/planning to control this via device factory?

I'm not sure, I added this to the factory because other "advanced" LUKS options like pbkdf are there too.

File: blivet/devicefactory.py [hunk 1]
```diff
@@ -1168,7 +1173,8 @@ def configure(self):
                 self.storage.format_device(member, get_format("luks",
                                                               min_luks_entropy=self.min_luks_entropy,
                                                               luks_version=self.luks_version,
-                                                              pbkdf_args=self.pbkdf_args))
+                                                              pbkdf_args=self.pbkdf_args,
+                                                              sector_size=self.luks_sector_size))
```
Comment (dwlehman) [inline blivet/devicefactory.py:1177]: You're passing `sector_size`  but the format class expects `luks_sector_size`.

Comment (dwlehman) [reply blivet/devicefactory.py:1177]: (Same throughout this file)

Comment (vojtechtrefny) [reply blivet/devicefactory.py]: Oops. Good catch. Fixed (and I also added the sector size change to devicefactory test case).

File: blivet/devicefactory.py [hunk 2]
```diff
@@ -339,6 +341,7 @@ def __init__(self, storage, **kwargs):
 
         self.luks_version = kwargs.get("luks_version") or crypto.DEFAULT_LUKS_VERSION
         self.pbkdf_args = kwargs.get("pbkdf_args", None)
+        self.luks_sector_size = kwargs.get("luks_sector_size", 0)
```
Comment (dwlehman) [inline blivet/devicefactory.py:344]: What will happen if someone passes `luks_sector_size=None`?

Comment (vojtechtrefny) [reply blivet/devicefactory.py]: Fixed.

**Pull request merged**